### PR TITLE
feat(plugin): Dato Table Editor Plugin

### PR DESCRIPTION
**Pull request by pbortnick**
🎟️ [Asana Task](https://app.asana.com/0/1200531701094631/1202533327481359/f)
🎨 [Design](https://www.figma.com/file/uxnWWpM14UcttJNvBadRfL/Pricing?node-id=205%3A7920)
^ Motivation for plugin

---

## Description

This PR includes a plugin for DatoCMS to enter our pricing/features data. It is built on top of a currently published [plugin](https://www.datocms.com/marketplace/plugins/i/datocms-plugin-table-editor?s=table) and includes different types of cell inputs including a wysiwyg editor with both heading and description, as well as a checkbox. The table also includes options to hide column headers and set rows to be collapsible. See [readme](https://github.com/hashicorp/web-platform-packages/pull/75/files#diff-5af5f7ff8d365754a8c4d3a2d3499d13b3b94e7e58a0ad3cd460fdd63218318f) for more information.

Once merged,  the plugin will be published and available in the [DatoCMS Plugin Marketplace](https://www.datocms.com/marketplace/plugins)

It is in the 🛹 stage and could use style updates, code refactoring, tests, etc.

### Testing instructions:
`cd packages/datocms-plugin-table`
`npm start`
Edit in [dato](https://io-marketing-pages.admin.datocms.com/editor/item_types/1926417/items/116196824). See **FOR TESTING ONLY - Features Table** section at the bottom of the page.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ The release process is handled mostly automatically via the changesets GitHub ac
 If you want to test your changes before merging, you can add a `release:canary` label to your pull request. If any changeset files are found, a release will be created and tagged with `canary`. You can then install the canary version elsewhere:
 
 ```
-npm instal @hashicorp/platform-package@canary
+npm install @hashicorp/platform-package@canary
```
Comment (pbortnick) [inline readme.md:57]: Just noticed this typo while working.

**Review by alexcarpenter — COMMENTED** [review]
Tested locally, and everything looks to be working good. Didn't review code throughly since I know this is built from the linked table-editor plugin that currently exists. 

One last nitpick is to align heading colors: 
CleanShot 2022-08-15 at 11 42 45@2x

**Comment by pbortnick** [comment]
> One last nitpick is to align heading colors

Thanks @alexcarpenter ! Updated in [this commit](https://github.com/hashicorp/web-platform-packages/pull/75/commits/3dbe7bfb2c8209cb874360657e383c090d1ea5ec)

**Review by alexcarpenter — APPROVED** [review]
👏🏼

File: packages/datocms-plugin-table/package.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "datocms-plugin-complex-table-editor",
```
Comment (alexcarpenter) [inline packages/datocms-plugin-table/package.json:2]: @pbortnick should this be `@hashicorp/datocms-plugin-complex-table-editor`?

Comment (pbortnick) [reply packages/datocms-plugin-table/package.json:2]: It has to be datocms-plugin-xxx https://www.datocms.com/docs/plugin-sdk/publishing-to-marketplace
Screen Shot 2022-08-15 at 12 15 34 PM

Comment (alexcarpenter) [reply packages/datocms-plugin-table/package.json:2]: @BRKalow ☝🏼 is there any issue in publishing without the `@hashicorp/` prefix with our npm publishing workflows?

Comment (BRKalow) [reply packages/datocms-plugin-table/package.json:2]: I think it should be okay! Let's give it a shot.

Comment (pbortnick) [reply packages/datocms-plugin-table/package.json:2]: If there is no changeset will anything be affected?

Comment (BRKalow) [reply packages/datocms-plugin-table/package.json:2]: @pbortnick Nope, without a changeset nothing will happen or be staged for publishing.

**Pull request merged**